### PR TITLE
Tdd/park container

### LINKED ISSUE #17: Testing for BannerIcons Components

### LINKED ISSUE #19: Testing for UserNav Component

### LINKED ISSUE #21: Testing for Banner Component

### LINKED ISSUE #28: Testing for SavedParks

### PR DESCRIPTION
**Pull request by baileydunning**
## Is this a fix or a feature?
Feature

## What is the change?
TDD for:
+ `Banner`
+ `BannerIcons`
+ `Footer`
+ `SavedParks`
+ `UserNav`

## Where should the reviewer start?
The testing files for the components listed above

## How should this be tested?
`npm test`

## Relevant Tickets
+ Closes #21 
+ Closes #19 
+ Closes #17 
+ Closes #28

@saraho1123
@jaypeasee
@baileydunning

### CODE REVIEW

File: src/BannerIcons/BannerIcons.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    userEvent.click(visitedBtn)
+    expect(addToVisited).toHaveBeenCalledWith('dena')
+    expect(addToVisited).toHaveBeenCalledTimes(1)
+
+    userEvent.click(visitedBtn)
+    expect(deleteFromVisited).toHaveBeenCalledWith('dena')
+    expect(deleteFromVisited).toHaveBeenCalledTimes(1)
+  })
+
+  it('should allow a user to add and delete a park to bucket list', () => {
```
Comment (saraho1123) [inline src/BannerIcons/BannerIcons.test.js:51]: Nice! I think I would have made double the amount of tests, but I honestly think it is better this way!

File: src/BannerIcons/BannerIcons.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ const BannerIcons: React.FC<BannerIconsProps> = props => {
         className="banner-btn-container">
         <img
           src={checkColor}
-          alt="not checked"
+          alt="visited"
```
Comment (saraho1123) [inline src/BannerIcons/BannerIcons.tsx:63]: I like this better!

File: src/BannerIcons/BannerIcons.test.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    userEvent.click(bucketListBtn)
+    expect(addToBucketList).toHaveBeenCalledWith('dena')
+    expect(addToBucketList).toHaveBeenCalledTimes(1)
+
```
Comment (jaypeasee) [inline src/BannerIcons/BannerIcons.test.js:55]: Nice job getting all of these to all render in the test and testing them after!

File: src/Footer/Footer.test.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    expect(visitedLink).toBeInTheDocument()
+    expect(bucketListLink).toBeInTheDocument()
+  })
+
+  it('should redirect to a new url', () => {
+    userEvent.click(visitedLink)
+    expect(history.location.pathname).toBe('/user/visited')
+    
+    userEvent.click(bucketListLink)
+    expect(history.location.pathname).toBe('/user/bucket-list')
```
Comment (jaypeasee) [inline src/Footer/Footer.test.js:33]: Great use of the history object

File: src/SavedParks/SavedParks.test.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import { screen, render } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import SavedParks from './SavedParks'
+import { Route, Router } from 'react-router-dom'
+import { createMemoryHistory } from 'history'
+import '@testing-library/jest-dom'
+
+describe('SavedParks', () => {
```
Comment (saraho1123) [inline src/SavedParks/SavedParks.test.js:8]: Thanks for showing me the nested describe blocks!

**Review by saraho1123 — APPROVED** [review]
Great test writing Bailey!

**Review by jaypeasee — APPROVED** [review]
This looks good. Only merge conflicts I see are in the ESLint and Package-lock.json which we can easily resolve

**Pull request merged**